### PR TITLE
Version Handling

### PR DESCRIPTION
**Pull request by dionysiusmarquis**
#### What type of PR is this?

/kind feature

#### What this PR does / why is it needed:

This PR aims to make version bumps to be handled like in other semver release projects.

#### Does this PR introduce a breaking change?

Major changes:
- `0.1.0` is the new first release version (following [initial-development-phase](https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase))
- only one version bump per release
- prerelease channel support (following [release-workflow/pre-releases](https://semantic-release.gitbook.io/semantic-release/recipes/release-workflow/pre-releases))

Minor changes:
- branch doesn't have to be present when set in config

#### Special notes for your reviewer:

To understand the way of version bumping it might be the easiest to go through this test scenario:
https://github.com/dionysiusmarquis/go-semver-release/blob/c290080718b630925e4a3ec9b0247766b7298bb3/cmd/release_test.go#L100

The new initial version is `0.1.0`. This will ever be the fist version ("ignoring" all commit rules for the first release). Version `0.0.0` is when it would be the first release but no commit with release rule is found.

If there are multiple rules applying since the last release the highest rule will be applied. E.g. the Version is `0.1.2` and there are multiple commits including `feat:`, `fix:`, `feat:` only the highest (`feat:` in this case) will be applied once - meaning the new version would be `0.2.0`.

Prereleases now have a build number. This is to run prereleases with a separate versioning. This way it's possible to apply fixes and features to the main channel with running prereleases. Prerelease versions are based on higher tier channels and get bumped if a higher tier version passes a prerelease version.

I plan to add more edge cases to the release cmd tests. But I thought I'd already post this pr to have a look at it and verify if this is a plausible direction.

### CODE REVIEW

**Comment by s0ders** [comment]
Hey! Thanks for you PR, I'm very busy at the moment but I'll take a closer look this weekend and get back to you here.

**Comment by dionysiusmarquis** [comment]
I found a way to utilize go-git `commitgraph` package to get a topo ordered commit history. Therefore I removed the commit walker implementation and corresponding tests again.

File: cmd/release.go [hunk 1]
```diff
@@ -82,6 +88,9 @@ func NewReleaseCmd(ctx *appcontext.AppContext) *cobra.Command {
 				logEvent.Bool("new-release", release)
 				logEvent.Str("version", semver.String())
 				logEvent.Str("branch", output.Branch)
+				if output.Error != nil {
+					logEvent.Err(output.Error)
```
[the remark this answers is not in the document]
Comment (dionysiusmarquis) [reply cmd/release.go:92]: I think both variants are valid. On the one hand, it might make integrations easier if all key value pairs are always present, on the other hand its redundant to provide values that are ignored when checking against the error first anyways. I‘ll change it to the way you think it makes more sense.
